pkg/lockfile/parse-go-lock.go

```diff
@@ -22,16 +22,25 @@ func ParseGoLock(pathToLockfile string) ([]PackageDetails, error) {
 		return []PackageDetails{}, fmt.Errorf("could not parse %s: %w", pathToLockfile, err)
 	}
 
-	packages := make([]PackageDetails, 0, len(parsedLockfile.Require))
+	packages := map[string]PackageDetails{}
 
 	for _, require := range parsedLockfile.Require {
-		packages = append(packages, PackageDetails{
+		packages[require.Mod.Path+"@"+require.Mod.Version] = PackageDetails{
 			Name:      require.Mod.Path,
 			Version:   strings.TrimPrefix(require.Mod.Version, "v"),
 			Ecosystem: GoEcosystem,
 			CompareAs: GoEcosystem,
-		})
+		}
 	}
 
-	return packages, nil
+	for _, replace := range parsedLockfile.Replace {
+		packages[replace.New.Path+"@"+replace.New.Version] = PackageDetails{
+			Name:      replace.New.Path,
+			Version:   strings.TrimPrefix(replace.New.Version, "v"),
+			Ecosystem: GoEcosystem,
+			CompareAs: GoEcosystem,
+		}
+	}
+
+	return pkgDetailsMapToSlice(packages), nil
```

Comment:
Should we move this out of https://github.com/google/osv-scanner/blob/4452f8cee25976ed656624dac8cb7c3cdc922436/pkg/lockfile/parse-npm-lock.go into a better filename? e.g. `utils.go` or something similar?

Reply:
Alternatively, we can just use https://pkg.go.dev/golang.org/x/exp/maps#Values

Reply:
If this was a language that had file-based scopings & imports (like Node) I'd agree, but since everything in Go is visible to the surrounding files in a folder/package, I generally take a light "rule of three" based approach unless there's another really compelling reason (like a big refactor or the specific function(s) are large and/or complex), to avoid bloating PRs etc.

Likewise with applying generics: since Go compiles to a static binary, and this is a pretty straightforward utility function that most likely won't change anytime soon, there's no real advantage (from what I understand) to switching right now since the same/similar compiled code will end up in the binary (and currently it might actually be a bit slower/bigger since generics are still being optimized).